### PR TITLE
chore: Update DataFusion

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
* keep up with dependencies, prepare for larger update (https://github.com/influxdata/idpe/issues/17990)
* Get new features (regr_* function) https://github.com/apache/arrow-datafusion/pull/7211

# Changes:
1. Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/99e2cd4b4082296b0e7f98b0fb122861c4f74a11

### CODE REVIEW

File: iox_query_influxql/src/plan/util_copy.rs [hunk 1]
```diff
@@ -52,273 +54,288 @@ where
         Some(replacement) => Ok(replacement),
         // No replacement was provided, clone the node and recursively call
         // clone_with_replacement() on any nested expressions.
```
Comment (alamb) [inline iox_query_influxql/src/plan/util_copy.rs:56]: this is somewhat annoying to keep this copy/pastad version of this code up to date -- I may try and make this function exported in DataFusion so we don't have to keep this copy updated)

Whitespace blind diff shows the actual difference is quite small: https://github.com/influxdata/influxdb_iox/pull/8447/files?w=1

File: iox_query_influxrpc/src/lib.rs [hunk 2]
```diff
@@ -1624,22 +1621,10 @@ impl AggExprs {
...
 
             let field_name = field.name;
-            agg_exprs.push(
-                Expr::GetIndexedField(GetIndexedField {
-                    expr: Box::new(selector.clone()),
-                    key: ScalarValue::from("value"),
-                })
-                .alias(field_name),
-            );
+            agg_exprs.push(selector.clone().field("value").alias(field_name));
```
Comment (alamb) [inline iox_query_influxrpc/src/lib.rs:1624]: cleaning this up uses the APIs from https://github.com/apache/arrow-datafusion/pull/7218

File: iox_query_influxql/src/plan/planner.rs [hunk 3]
```diff
@@ -1198,11 +1198,8 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
                     aggr_exprs[0] = selector_new.clone();
 
                     for (idx, struct_name, out_alias) in fields_to_extract {
-                        select_exprs[idx] = Expr::GetIndexedField(GetIndexedField {
-                            expr: Box::new(selector_new.clone()),
-                            key: ScalarValue::Utf8(Some(struct_name)),
-                        })
-                        .alias(out_alias);
+                        select_exprs[idx] =
+                            selector_new.clone().field(struct_name).alias(out_alias);
```
Comment (NGA-TRAN) [inline iox_query_influxql/src/plan/planner.rs:1202]: Shorter call 👍

**Review by NGA-TRAN — APPROVED** [review]
Nice

**Review by appletreeisyellow — APPROVED** [review]
Thank you for keeping IOx updated with DataFusion! 🚀

**Pull request merged**